File: Cocoa Programming/Cocoa Pr L69 ObjC/Cocoa Pr L69 ObjC/Window.m
//
//  Window.m
//  Cocoa Pr L69 ObjC
//
//  Created by Michele Galvagno on 02/09/22.
//

#import "Window.h"

@implementation Window

- (instancetype)initWithContentRect:(NSRect)contentRect styleMask:(NSWindowStyleMask)style backing:(NSBackingStoreType)backingStoreType defer:(BOOL)flag {
    self = [super initWithContentRect:contentRect styleMask:NSWindowStyleMaskFullSizeContentView backing:NSBackingStoreBuffered defer:YES];
    
    self.movableByWindowBackground = YES;
    self.backgroundColor = [NSColor systemGrayColor];
    NSLog(@"Window class initialised!");
    
    return self;
}



@end

File: Cocoa Programming/Cocoa Pr L69 NSWindowController/Cocoa Pr L69 NSWindowController/Window.swift
//
//  Window.swift
//  Cocoa Pr L69 NSWindowController
//
//  Created by Michele Galvagno on 28/08/22.
//

import Cocoa

class Window: NSWindow {
    override init(contentRect: NSRect, styleMask style: NSWindow.StyleMask, backing backingStoreType: NSWindow.BackingStoreType, defer flag: Bool) {
        super.init(contentRect: contentRect, styleMask: [.fullSizeContentView], backing: .buffered, defer: true)
        isMovableByWindowBackground = true
        backgroundColor = NSColor.systemGray
    }
}
